AK/Checked.h

```diff
@@ -199,6 +199,28 @@ class Checked {
         m_value /= other;
     }
 
+    constexpr void saturating_sub(T other)
```

Comment:
Note:
There is a Fixed-Point c extension that GCC supports, [that  adds the
"_Sat" type modifier](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1169.pdf), that may look like what we want here.
I could not find anything about clang supporting it though, and the the format is slighty different